nltk/tokenize/treebank.py

```diff
@@ -163,7 +163,26 @@ def span_tokenize(self, text):
             True
 
         """
-        tokens = self.tokenize(text)
+        raw_tokens = self.tokenize(text)
+
+        # Convert converted quotes back to original double quotes
+        # Do this only if original text contains double quote(s)
+        if '"' in text:
+            # Find double quotes and converted quotes
+            matched = [m.group() for m in re.finditer(r'[(``)(\'\')(")]+', text)]
+            
+            # Replace converted quotes back to double quotes
+            index = 0
+            tokens = []
+            for tok in raw_tokens:
+                if tok in ['"', "``", "''"]:
+                    tokens.append(matched[index])
+                    index += 1
+                else:
+                    tokens.append(tok)
```

Comment:
If we use `matched.pop(0)`, we can avoid the index counting =)

How about this?

```python
tokens = [matched.pop(0) if tok in ['"', "``", "''"] else tok for tok in raw_tokens]
```

Essentially:

```python
tokens = []
for tok in raw_tokens:
    if tok in ['"', "``", "''"]:
        tokens.append(matched.pop(0))
    else:
        tokens.append(tok)
```